src/safeposix/syscalls/sys_calls.rs

```diff
@@ -161,15 +207,18 @@ impl Cage {
                             }
                         }
                     }
+                    // If the fd is linked to a pipe increment the ref count of the pipe
                     Pipe(pipe_filedesc_obj) => {
                         pipe_filedesc_obj.pipe.incr_ref(pipe_filedesc_obj.flags)
                     }
+                    // If the fd is linked to a socket increment the ref count of the socket
                     Socket(socket_filedesc_obj) => {
-                        // checking whether this is a domain socket
+                        // Check if it is a domain socket
                         let sock_tmp = socket_filedesc_obj.handle.clone();
                         let mut sockhandle = sock_tmp.write();
                         let socket_type = sockhandle.domain;
                         if socket_type == AF_UNIX {
+                            // Increment the appropriate reference counter of the correct socket
```

Comment:
Why are we only checking for socket type of `AF_UNIX` and a small comment about each socket type would be great for understanding.

Reply:
Resolved.